### PR TITLE
[Magento 2.2] Upload fails in image uploader

### PR DESCRIPTION
**Pull request by progreg**
### Description
The issue can be reproduced only in case if `fileinfo` module is disabled or not installed. Problem is that mime type of uploaded file would be resolved by `tmp` file. And in case of disabled `fileinfo` module `Uploader` will try to resolve mime type by file extension that always equals null for `tmp` files

### Fixed Issues

1. magento/magento2#16531: [Magento 2.2.5] Upload fails in image uploader

### Manual testing scenarios (*)

1. Ensure that `fileinfo` module is disabled  
2. Open Cms Block edit page
3. Select the image uploader ("Insert Image" button or the one in the editor)
4. Select an image from your hd and upload it.

#### Expected result
An added image in the gallery

#### Actual result
An error is shown: "File validation failed." No image is uploaded.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @progreg. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.2-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by miguelbalparda** [comment]
Hi @progreg! Thank you for your contribution. Care to check the failing tests? They seem related to the new code. Thanks!

**Comment by progreg** [comment]
> Hi @progreg! Thank you for your contribution. Care to check the failing tests? They seem related to the new code. Thanks!

Hi @miguelbalparda, I will check, thanks

**Comment by progreg** [comment]
@miguelbalparda, seems all checks have successfully passed. If need more explanation why I did such changes just tell me. thanks

File: lib/internal/Magento/Framework/File/Mime.php [hunk 1]
```diff
@@ -80,10 +80,11 @@ class Mime
      * Get mime type of a file
      *
      * @param string $file
+     * @param string $default
      * @return string
      * @throws \InvalidArgumentException
      */
-    public function getMimeType($file)
+    public function getMimeType($file, $default = 'application/octet-stream')
```
Comment (sidolov) [inline lib/internal/Magento/Framework/File/Mime.php]: Hi @progreg , unfortunately, we can't change public methods signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/File/Mime.php (hunk 1 above)
Comment (progreg) [reply lib/internal/Magento/Framework/File/Mime.php]: Hi @sidolov, I've added changes according to Backward Compatible Development Guide, I hope all was done in the right way

**Comment by progreg** [comment]
Hi @sidolov, can you check this PR again?

**Comment by orlangur** [comment]
@progreg we cannot process PRs from an `unknown repository`. Also, first fix in `2.3-develop` should be processed.

**Pull request closed without merging**